spec/integration/repository_spec.rb

```diff
@@ -0,0 +1,89 @@
+describe PackageProvider::Repository do
+  PackageProvider::Repository.temp_prefix = "pp_integration_tests_#{rand(1000)}"
+  persist_folders_prefix = 'pp_integration_per'
+
+  let(:fake_remote_repo_dir) do
+    File.join(PackageProvider.root, 'spec', 'factories', 'testing-repo')
+  end
+  let(:repo) do
+    PackageProvider::Repository.new(fake_remote_repo_dir)
+  end
+
+  after(:each) do
+    repo && repo.destroy
+  end
+
+  describe '#initialize' do
+    it 'makes local copy' do
+      expect(
+        PackageProvider::Spec::Helpers.test_git_status(repo.repo_root)
+      ).to be true
+    end
+  end
+
+  describe '#clone' do
+    let(:dest_dir) do
+      PackageProvider::Spec::Helpers.get_temp_dir_name(persist_folders_prefix)
```

Comment:
see bellow. Generally find & replace calling all calling `PackageProvider::Spec::Helpers.<method>` to only method. For hepers use `module` and include them.